knownvalue/bool.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package knownvalue
+
+import (
+	"fmt"
+	"strconv"
+)
+
+var _ Check = BoolValue{}
+
+// BoolValue is a Check for asserting equality between the value supplied
+// to BoolValueExact and the value passed to the CheckValue method.
+type BoolValue struct {
```

Comment:
Sounds good to me.

The known value check types have all be modified so that they are camel-cased versions of the exported functions, for example:

- `boolValueExact` => `BoolValueExact`
- `listElementsExact => `ListElementsExact`
- `listValueExact` => `ListValueExact`
- `listValuePartial` => `ListValuePartial` (note that the constructor for the know value partial types has changed from `<List|Map|Object|Set>ValuePartialMatch` to `<List|Map|Object|Set>ValuePartial`)